specification/azureactivedirectory/resource-manager/Microsoft.Aadiam/preview/2020-07-01-preview/azureADMetrics.json

```diff
@@ -0,0 +1,432 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2020-07-01-preview",
+    "title": "azureactivedirectory",
+    "description": "Metrics and Alerts for Azure Active Directory."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/microsoft.aadiam/azureADMetrics/{azureADMetricsName}": {
+      "put": {
+        "tags": [
+          "azureADMetrics"
+        ],
+        "operationId": "azureADMetrics_Create",
+        "x-ms-long-running-operation": true,
```

Comment:
x-ms-long-running-operation implies that your PUT API is an async operation.  If you need it to be async then you need to have a "properties.provisioningState" property within your resource.  Right now you don't.  See https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creatingupdating-using-put for more info

Reply:
can provisioningState be added directly to the azureADMetricsConfig definition without having the "additional properties not allowed" error?

Reply:
No, it needs to be inside an object named "properties" within azureAdMetricsConfig

Reply:
{
            code: 'OBJECT_ADDITIONAL_PROPERTIES',
            params: [ 'properties' ],
            message: 'Additional properties not allowed: properties',
            path: [],
            title: '#/definitions/azureADMetricsConfig',
            description: 'AzureADMetrics resource.',
            position: {
              line: 325,
              column: 29,
              directives: { RequiredPropertiesMissingInResourceModel: '.*' }
            },

Reply:
^ This is what I am getting when I am putting the provisioningState within a properties object.

Reply:
I'm not exactly sure why this is happening, because PrivateLink also has additional properties under "properties" and this is in an identical format.